File: week1/problems/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int add_coins(int total, int change);

int main(void)
{
    float dollars;
    // Ask user for change owed
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars <= 0);

    int coins = 0;
    int cents = round(dollars * 100);
    int quarter = 25, dime = 10, nickle = 5, penny = 1;

    // Add quarters and update total
    int quarters = add_coins(cents, quarter);
    cents = cents - (quarters * quarter);

    // Add dimes and update total
    int dimes = add_coins(cents, dime);
    cents = cents - (dimes * dime);

    // Add nickles and update total
    int nickles = add_coins(cents, nickle);
    cents = cents - (nickles * nickle);

    // Add pennies
    int pennies = add_coins(cents, penny);

    coins = quarters + dimes + nickles + pennies;

    printf("%i\n", coins);
}

int add_coins(int total, int change)
{
    int coins = 0;

    while (total >= change)
    {
        total -= change;
        coins++;
    }

    return coins;
}

File: week6/problems/cash.py
from cs50 import get_float
from math import floor


def main():
    # Ask user for positive change owed
    while True:
        dollars = get_float("Change owed: ")
        if dollars > 0:
            break

    cents = round(dollars * 100)

    # Calculate quarters, dimes, nickels and pennies
    quarters, remain = calc(cents, 25)
    dimes, remain = calc(remain, 10)
    nickels, remain = calc(remain, 5)
    pennies, remain = calc(remain, 1)

    # Print total number of coins
    print(quarters + dimes + nickels + pennies)


# Calculate number of change and remaining cash
def calc(total, change):
    coins = floor(total / change)
    remain = total % change

    return coins, remain


if __name__ == "__main__":
    main()